test/controllers/thesis_controller_test.rb

```diff
@@ -769,4 +769,122 @@ def attach_files_to_records(tr, th)
     follow_redirect!
     assert_select '.alert-banner.success', text: 'The theses you selected have been added to the publication queue. Status updates are not immediate.', count: 1
   end
+
+  # ~~~~~~~~~~~~~~~~~~~~~ proquest export preview ~~~~~~~~~~~~~~~~~~~~~
+  test 'anonymous users get redirected if they load the proquest export preview list' do
+    get thesis_proquest_export_preview_path
+    assert_response :redirect
+    assert_redirected_to '/login'
+  end
+
+  test 'basic users get redirected if they load the proquest export preview list' do
+    sign_in users(:basic)
+    get thesis_proquest_export_preview_path
+    assert_redirected_to '/'
+    follow_redirect!
+    assert_select 'div.alert', text: 'Not authorized.', count: 1
+  end
+
+  test 'submitters get redirected if they load the proquest export preview list' do
+    sign_in users(:transfer_submitter)
+    get thesis_proquest_export_preview_path
+    assert_redirected_to '/'
+    follow_redirect!
+    assert_select 'div.alert', text: 'Not authorized.', count: 1
+  end
+
+  test 'processors can load the proquest export preview list' do
+    sign_in users(:processor)
+    get thesis_proquest_export_preview_path
+    assert_response :success
+  end
+
+  test 'thesis admins can load the proquest export preview list' do
+    sign_in users(:thesis_admin)
+    get thesis_proquest_export_preview_path
+    assert_response :success
+  end
+
+  test 'admins can load the proquest export preview list' do
+    sign_in users(:admin)
+    get thesis_proquest_export_preview_path
+    assert_response :success
+  end
+
+  test 'proquest export cannot be initiated with no eligible theses' do
+    sign_in users(:processor)
+    Thesis.destroy_all
```

Comment:
Oh, good call.

Reply:
Ah, that's right. We can't use `delete_all` because we need  the `dependent: :destroy` callbacks, or else these tests will throw foreign key constraint exceptions.

Reply:
🙈